onnxruntime/core/providers/dml/DmlExecutionProvider/src/GraphDescBuilder.cpp

```diff
@@ -139,56 +139,79 @@
+            for (auto& operatorGraphInputEdge : graphNodeCreateInfo.inputEdges)
+            {
+                //auto& operatorGraphInputEdge = graphNodeCreateInfo.inputEdges[inputIndex];
+                // skip the optional input of a DML operator API.
+                /*if (operatorGraphInputEdge.GraphInputIndex == std::numeric_limits<uint32_t>::max())
+                {
+                    continue;
+                }*/
 
-                const onnxruntime::NodeArg* arg = node.InputDefs()[kernelInputIndex];
+                // operatorGraphInputEdge.GraphInputIndex will be the ONNX input index.
+                const onnxruntime::NodeArg* arg = node.InputDefs()[operatorGraphInputEdge.GraphInputIndex];
 
                 if (arg->Exists())
                 {
-                    auto iter = nameToFusedNodeInputIndex.find(arg->Name());
+                    auto iter = nameToDmlFusedNodeInputIndex.find(arg->Name());
+                    uint32_t mainGraphNodeIndex = operatorGraphNodeIndexToMainGraphNodeIndexMap[operatorGraphInputEdge.ToNodeIndex];
 
-                    if (iter != nameToFusedNodeInputIndex.end())
+                    if (iter != nameToDmlFusedNodeInputIndex.end())
                     {
                         // This is a graph input
 
-                        const uint32_t fusedNodeInputIndex = iter->second;
+                        const uint32_t dmlFusedNodeInputIndex = iter->second;
 
                         DML_INPUT_GRAPH_EDGE_DESC edge = {};
-                        edge.GraphInputIndex = fusedNodeInputIndex;
-                        edge.ToNodeIndex = nodeIndex;
-                        edge.ToNodeInputIndex = inputIndex;
+                        edge.GraphInputIndex = dmlFusedNodeInputIndex;
+                        edge.ToNodeIndex = mainGraphNodeIndex;
+                        edge.ToNodeInputIndex = operatorGraphInputEdge.ToNodeInputIndex;  // ?? might need to point inputIndex
                         graphInputEdges.push_back(edge);
 
                         // If this is a constant input, set the appropriate flags on the desc
-                        if (fusedNodeInputIndex < isConstGpuGraphInputCount && isConstGpuGraphInput[fusedNodeInputIndex])
+                        if (isNodeAsOpDesc && 
+                            dmlFusedNodeInputIndex < isConstGpuGraphInputCount && 
+                            isConstGpuGraphInput[dmlFusedNodeInputIndex])
                         {
-                            DmlBufferTensorDesc* tensorDesc = inputTensorDescs[inputIndex];
-
+                            std::vector<DmlBufferTensorDesc*> toNodeInputTensorDescs = 
+                                (*graphNodeCreateInfo.nodesAsOperatorDesc)[operatorGraphInputEdge.ToNodeIndex]->GetInputTensors();
+                            DmlBufferTensorDesc* tensorDesc = toNodeInputTensorDescs[operatorGraphInputEdge.ToNodeInputIndex];  // ?? might need to point inputIndex
```

Comment:
If you ever find yourself splitting a line like this, it may indicate you're doing too much in a single statement, and doing too much in a single statement can make debugging more complicated, stepping in and out of several little functions along the way. Consider assigning to a temporary a fragment, like `auto& graphInputNode =(*graphNodeCreateInfo.nodesAsOperatorDesc)[operatorGraphInputEdge.ToNodeIndex]`.